candidate/models.py

```diff
@@ -0,0 +1,93 @@
+from __future__ import unicode_literals
+from django.db import models
+from django.utils.encoding import python_2_unicode_compatible
+
+from ballot_measure.models import BallotMeasureChoice
+
+
+class SocialMediaModel(models.Model):
+    class Meta:
+        abstract = True
+
+    photo_url = models.ImageField(blank=True)
+    website_url = models.URLField(
+        help_text='URL for the official website.', blank=True)
+    facebook_url = models.URLField(
+        help_text='URL for the Facebook page.', blank=True)
+    twitter_url = models.URLField(
+        help_text='URL for the Twitter page.', blank=True)
+
+
+@python_2_unicode_compatible
+class Party(SocialMediaModel):
+    """
+    """
+    name = models.CharField(
+        max_length=255, help_text='The party name.')
+
+    def __str__(self):
+        return "The %s party" % self.name
+
+
+@python_2_unicode_compatible
+class Person(SocialMediaModel):
+    """
+    """
+    first_name = models.CharField(
+        max_length=255, help_text='The candidate\'s first name.')
+    middle_name = models.CharField(
+        max_length=255, blank=True, null=True,
+        help_text='The candidate\'s middle name.')
+    last_name = models.CharField(
+        max_length=255, blank=True, null=True,
+        help_text='The candidate\'s last name.')
+    party = models.ForeignKey('Party', blank=True, null=True)
+
+    def __str__(self):
+        return "%s, %s%s" % (
+            self.last_name, self.first_name,
+            '' if self.middle_name is None else ' %s.' % self.middle_name[0])
+
+
+@python_2_unicode_compatible
+class Office(models.Model):
+    """
+    """
+    name = models.CharField(
+        max_length=255, help_text='The office name.')
+    description = models.CharField(
+        max_length=1024, help_text='The office description.')
+    locality = models.ForeignKey('locality.Locality')
+
+    def __str__(self):
+        return "Office for %s in %s" % (
+            self.name, self.locality.dereference())
+
+
+@python_2_unicode_compatible
+class Election(models.Model):
```

Comment:
Yeah, I'm not sure about this either. On one hand, I don't really see the benefit of normalizing the event of election day into an `Election` object. I don't think we'll need to build any features that rely on that normalization, such as a list of all elections happening November 6th, 2016. And, it will just result in us having to join more tables in order to render a `Locality`.

On the other hand, this implementation of `Election` as a join table between `Office` and `BallotMeasure` is not exactly what I would expect when thinking of an election either. I think I like `OfficeElection` as an alternative name, or maybe a `BallotMeasure` can just relate directly to an `Office` (optionally, of course. Referendums have no office).